nexus/tests/integration_tests/updates.rs

```diff
@@ -0,0 +1,289 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+// TODO(iliana):
+// - refactor `test_update_end_to_end` into a test setup function
+// - test that an unknown artifact returns 404, not 500
+// - tests around target names and artifact names that contain dangerous paths like `../`
+
+use chrono::{Duration, Utc};
+use dropshot::test_util::LogContext;
+use dropshot::{
+    endpoint, ApiDescription, HttpError, HttpServerStarter, Path,
+    RequestContext,
+};
+use http::{Method, Response, StatusCode};
+use hyper::Body;
+use nexus_test_utils::{load_test_config, test_setup, test_setup_with_config};
+use omicron_common::api::internal::nexus::UpdateArtifactKind;
+use omicron_nexus::config::UpdatesConfig;
+use omicron_nexus::updates::{ArtifactsDocument, UpdateArtifact};
+use ring::pkcs8::Document;
+use ring::rand::{SecureRandom, SystemRandom};
+use ring::signature::Ed25519KeyPair;
+use schemars::JsonSchema;
+use serde::Deserialize;
+use std::collections::HashMap;
+use std::convert::TryInto;
+use std::fmt::{self, Debug};
+use std::fs::File;
+use std::io::Write;
+use std::num::NonZeroU64;
+use std::path::PathBuf;
+use std::sync::Arc;
+use tempfile::{NamedTempFile, TempDir};
+use tough::editor::signed::{PathExists, SignedRole};
+use tough::editor::RepositoryEditor;
+use tough::key_source::KeySource;
+use tough::schema::{KeyHolder, RoleKeys, RoleType, Root};
+use tough::sign::Sign;
+
+// If you're getting a double-panic here, run the test with
+// `cargo test -- --nocapture test_update_end_to_end` to get output.
+// `dropshot::HttpServer`'s `Drop` implementation panics.
+#[tokio::test]
+async fn test_update_end_to_end() {
+    let mut config = load_test_config();
+
+    // build the TUF repo
+    let rng = SystemRandom::new();
+    let tuf_repo = new_tuf_repo(&rng);
+
+    // serve it over HTTP
+    let dropshot_config = Default::default();
+    let mut api = ApiDescription::new();
+    api.register(static_content).unwrap();
+    let context = FileServerContext { base: tuf_repo.path().to_owned() };
+    let logctx = LogContext::new("test_update_end_to_end", &config.log);
+    let server =
+        HttpServerStarter::new(&dropshot_config, api, context, &logctx.log)
+            .unwrap()
+            .start();
+    let local_addr = server.local_addr();
+    tokio::spawn(async move { server.await });
+
+    // stand up the test environment
+    config.updates = Some(UpdatesConfig {
+        trusted_root: tuf_repo.path().join("metadata").join("1.root.json"),
+        default_base_url: format!("http://{}/", local_addr),
+    });
+    let cptestctx =
+        test_setup_with_config("test_update_end_to_end", &mut config).await;
+    let client = &cptestctx.external_client;
+
+    // FIXME: stand up an HTTP server rooted at `tuf_repo` and configure nexus
```

Comment:
Is this not what you're already doing? (or is this just a leftover comment)

Reply:
leftover comment oops